internal/isoeditor/rhcos.go

```diff
@@ -352,6 +360,20 @@ func tempFileName(baseDir string) (string, error) {
 	return path, nil
 }
 
+func writeAt(b []byte, offset int64, clusterISOPath string) error {
+	iso, err := os.OpenFile(clusterISOPath, os.O_WRONLY, 0o664)
```

Comment:
Right, good catch, I'll add iso.Close.
Regarding the header key, we could just make it public for reusability, it's just nicer to include the validation in ParseOffsetInfo func, would that be useful enough?
The writeAt is just an helper, might not be helpful in the full-iso scenario indeed.

Reply:
> ParseOffsetInfo func, would that be useful enough?

It will certainly _work_. My opinion was more like this...it probably isn't the greatest for `upload_iso.go` to "know" which bytes it needs in order to get the embed area. However, it's using s3 tricks to get a specific subset. I think it's a reasonable responsibility of the `isoeditor` package to know how to read the embed area without telling you what key to use.

It's hard for me to clearly delineate the separation of responsibilities between these packages as the waters are kind of murky. I do believe the less users of the `isoeditor` need to know about the implementation...the better.

Reply:
Sure I get your point, so I would simplify ParseOffsetInfo and add a separate helper for validation.

Reply:
I don't see `iso` getting closed.

Reply:
done